util/codegen/codegen_test.go

```diff
@@ -174,3 +176,88 @@ func TestGenericContainsPointers(t *testing.T) {
 		})
 	}
 }
+
+func TestAssertStructUnchanged(t *testing.T) {
+	type args struct {
+		t      *types.Struct
+		tname  string
+		params *types.TypeParamList
+		ctx    string
+		it     *ImportTracker
+	}
+
+	// package t1 with a struct T1 with two fields
+	p1 := types.NewPackage("t1", "t1")
+	p1.Scope().Insert(types.NewTypeName(0, p1, "T1", types.NewStruct([]*types.Var{
+		types.NewField(0, nil, "P1", types.Typ[types.Int], false),
+		types.NewField(0, nil, "P2", types.Typ[types.String], false),
+	}, nil)))
+
+	// package t2 with a struct T2 with a named field T1 T1 and two other fields
+	p2 := types.NewPackage("t2", "t2")
+	p2.Scope().Insert(types.NewTypeName(0, p2, "T1", types.NewNamed(types.NewTypeName(0, p1, "T1", nil), nil, nil)))
```

Comment:
I think this is somewhat equivalent to `type T1 = t1.T1` and is unnecessary.
Let's remove lines 198 and 207.